### PR TITLE
Upgrade to Log4j 2.8.2

### PR DESCRIPTION
**Pull request by jasontedor**
This commit upgrades the Log4j dependencies from version 2.7 to version 2.8.2. This release includes a fix for a case where Log4j could lose exceptions in the presence of a security manager.

### CODE REVIEW

**Review by dadoonet — APPROVED** [review]

**Comment by dadoonet** [comment]
Note that I think the Java guide should be updated as well?

File: docs/java-api/index.asciidoc [hunk 1]
```diff
@@ -44,12 +44,12 @@ You need to also include Log4j 2 dependencies:
 <dependency>
     <groupId>org.apache.logging.log4j</groupId>
     <artifactId>log4j-api</artifactId>
-    <version>2.7</version>
+    <version>2.8.2</version>
```
Comment (olcbean) [inline docs/java-api/index.asciidoc]: It should be possible to define an attribute / variable for the log4j version and change it only at one place

Comment (jasontedor) [inline docs/java-api/index.asciidoc]: Not between docs and outside the docs. The docs are built independently, do not use Gradle, and do not consume the entire repository.

Comment (olcbean) [inline docs/java-api/index.asciidoc]: I meant in the doc itself that you can define it (not between the src and doc). So at least in the doc you can reuse the same variable.

Comment (jasontedor) [inline docs/java-api/index.asciidoc]: I think for one value in a single file it's not worth the indirection.

**Comment by jasontedor** [comment]
Thanks for the helpful review @dadoonet.

**Pull request merged**